pkg/api/v1alpha1/cluster_webhook.go

```diff
@@ -54,19 +55,22 @@ var _ webhook.Validator = &Cluster{}
 // ValidateCreate implements webhook.Validator so a webhook will be registered for the type
 func (r *Cluster) ValidateCreate() error {
 	clusterlog.Info("validate create", "name", r.Name)
-	if r.IsReconcilePaused() {
-		clusterlog.Info("cluster is paused, so allowing create", "name", r.Name)
-		return nil
-	}
-	if !features.IsActive(features.FullLifecycleAPI()) {
-		return apierrors.NewBadRequest("Creating new cluster on existing cluster is not supported")
+	var allErrs []error
+	if err := r.Validate(); err != nil {
+		allErrs = append(allErrs, fmt.Errorf("invalid cluster: %v ", err))
 	}
-	if r.IsSelfManaged() {
-		return apierrors.NewBadRequest("Creating new cluster on existing cluster is not supported")
+
+	if !r.IsReconcilePaused() {
+		if r.IsSelfManaged() {
+			allErrs = append(allErrs, fmt.Errorf("creating new cluster on existing cluster is not supported for self managed clusters"))
+		} else if !features.IsActive(features.FullLifecycleAPI()) {
+			allErrs = append(allErrs, fmt.Errorf("creating new managed cluster on existing cluster is not supported"))
+		}
 	}
 
-	if err := validateCNIPlugin(r.Spec.ClusterNetwork); err != nil {
-		return apierrors.NewBadRequest(err.Error())
+	if len(allErrs) > 0 {
+		aggregate := utilerrors.NewAggregate(allErrs)
+		return apierrors.NewBadRequest(aggregate.Error())
```

Comment:
What would you think about doing something similar to this:
https://github.com/kubernetes/apimachinery/blob/master/pkg/api/errors/errors.go#L292

Reply:
That looks like a good idea.

We would need to combine the [NewInvalid](https://github.com/kubernetes/apimachinery/blob/master/pkg/api/errors/errors.go#L282) and the [BadRequest](https://github.com/kubernetes/apimachinery/blob/master/pkg/api/errors/errors.go#L313) errors into the Status Error, which the library does not support at the moment. We could write our own logic for this, and we might use it for future PRs for returning similar errors from webhooks as well. 

Created this task https://github.com/aws/eks-anywhere/issues/2813.